dateparser/parser.py

```diff
@@ -346,16 +346,20 @@ def _get_date_obj(self, token, directive):
         return strptime(token, directive)
 
     def _results(self):
-        if self.settings.STRICT_PARSING:
-            errors = []
-            if not self.day:
-                errors.append('Day')
-            if not self.month:
-                errors.append('Month')
-            if not self.year:
-                errors.append('Year')
+        missing = []
+        if not self.day:
+            missing.append('day')
+        if not self.month:
+            missing.append('month')
+        if not self.year:
+            missing.append('year')
+
+        if self.settings.STRICT_PARSING and missing:
+            raise ValueError('%s not found in the date string' % ', '.join(missing))
```

Comment:
:lipstick: “day, month not found in the date string” feels weird to me. Same applies below.

```suggestion
            raise ValueError('Fields missing from the date string: %s' % ', '.join(missing))
```

Reply:
This message is not shown, in fact, before my changes it wasn't separated by commas, creating a weird/wrong message like "DayMonthYear not found in the date string". 

However, I decided to follow your advice and I changed the code to use a better message. Thanks @Gallaecio!